src/MongoDB.EntityFrameworkCore/Storage/MongoClientWrapper.cs

```diff
@@ -155,19 +155,48 @@ public async Task<bool> DeleteDatabaseAsync(CancellationToken cancellationToken
     /// <inheritdoc/>
     public bool DatabaseExists()
     {
-        using var cursor = _client.ListDatabaseNames(BuildListDbNameOptions());
-        return cursor.Any();
+        if (_useDatabaseNameFilter)
```

Comment:
It's risky to cache information like this because if they upgrade the server without restarting the application the application is stuck forever thinking that `_useDsatabaseNameFilter == false`.

If we don't care about that then this is fine.